torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -412,9 +414,15 @@ TensorPipeEntry tensorpipeSerialize(const Message& rpcMessage) {
                          std::move(copiedTensors)};
 }
 
-Message tensorpipeAllocateMessage(const tensorpipe::Message& tpMessage) {
+Message tensorpipeAllocateMessage(tensorpipe::Message& tpMessage) {
   // Payload, message type and message id
-  std::vector<char> payload(tpMessage.length);
+  TORCH_INTERNAL_ASSERT(
+      tpMessage.payloads.size() == 1,
+      "message expected to contain 1 payload, whereas it contained ",
+      tpMessage.payloads.size(),
+      " payloads");
+  std::vector<char> payload(tpMessage.payloads[0].length);
+  tpMessage.payloads[0].data = (uint8_t*)(payload.data());
```

Comment:
@lw If we look at this one, the check is just valid for current interface usage. When start to handle duplicated tensor storage and put all tensors' metadata to payload[1], this will break. Does that mean this should be removed..?

Reply:
This check is being performed _inside_ the function, not by any of its users or by the tests.